internal/query/relations.go

```diff
@@ -572,6 +572,75 @@ func (e *engine) listRoleResourceActions(ctx context.Context, role types.Resourc
 	return resourceActions, nil
 }
 
+// GetRole gets the role with it's actions.
+func (e *engine) GetRole(ctx context.Context, roleResource types.Resource, queryToken string) (types.Role, error) {
+	var (
+		resActions map[types.Resource][]string
+		err        error
+	)
+
+	for _, resType := range e.schemaRoleables {
+		resActions, err = e.listRoleResourceActions(ctx, roleResource, resType.Name, queryToken)
+		if err != nil {
+			return types.Role{}, err
+		}
+
+		// roles are only ever created for a single resource, so we can break after the first one is found.
+		if len(resActions) != 0 {
+			break
+		}
+	}
+
+	if len(resActions) > 1 {
+		e.logger.Warnw("role is assigned to more than one resource", "role.id", roleResource.ID.String())
+	}
```

Comment:
IMO it'd probably be better for us to return an error here, since this is something that shouldn't happen.